ontology_api/role/role_service.py

```diff
@@ -0,0 +1,116 @@
+from datetime import datetime
+from owlready2 import FunctionalProperty
+from role.role_model import ObjectPropertyRoleModelIn, DataTypePropertyRoleModelIn
+from role.role_model import ObjectPropertyRoleModelOut, DataTypePropertyRoleModelOut
+from model.model_service import ModelService
+
+
+class RoleService:
+    """
+        Object to handle logic of role requests
+    """
+
+    def __init__(self):
+        self.model_service = ModelService()
+
+    def add_object_property_role(self, model_id: int, model_in: ObjectPropertyRoleModelIn):
+        """
+                Create an instance of an object property in the given model
+        """
+        onto = self.model_service.load_ontology(model_id)
+
+        if onto is None:
+            return ObjectPropertyRoleModelOut(errors=f"Model with id {model_id} not found")
+
+        onto_property = onto[model_in.role_name]
+
+        if onto_property is None:
+            onto.destroy()
+            return ObjectPropertyRoleModelOut(errors=f"Property {model_in.role_name} not found")
+
+        property_domain = onto_property.domain
+        property_range = onto_property.range
+        src_instance = onto.search_one(type=property_domain, iri=f"*{model_in.src_instance_name}")
+
+        if src_instance is None:
+            onto.destroy()
+            return ObjectPropertyRoleModelOut(errors=f"Instance {model_in.src_instance_name} not found")
+
+        dst_instance = onto.search_one(type=property_range, iri=f"*{model_in.dst_instance_name}")
+
+        if dst_instance is None:
+            onto.destroy()
+            return ObjectPropertyRoleModelOut(errors=f"Instance {model_in.dst_instance_name} not found")
+        code = f"""
+if FunctionalProperty in onto_property.is_a:
+    src_instance.{model_in.role_name} = dst_instance
+else:
+    if src_instance.{model_in.role_name} is not None:
+        src_instance.{model_in.role_name}.append(dst_instance)
+    else:
+        src_instance.{model_in.role_name} = [dst_instance]                
+"""
+        exec(code)
```

Comment:
I don't like this code at all. 
What is the `src_instance`? Is it a class? A dict? If it's a class you can probably use `getattr`... 
`exec` is almost never a good idea...